### DISCUSSION TITLE
Construction timelapse with daily auto upload

### DISCUSSION THREAD

**Discussion by serek1107** [gopro/labs#173] [Q&A]
Hi, 

I would like to generate a settings for making a 3 months construction timelapse with daily auto upload to the cloud. 
Camera will be powered by external battery, maybe AC-power if available and connected to wifi signal. I would like to have an option to auto upload daily photos without removing them from the SD card. Timelapse photos will be taken from 8 am to 4 pm, every 30 min, and camera should be shut off between shots. I would like for daily auto upload to be after the last shot of the day. 
Is there a possibility to create such settings?

**Comment by dnewman-gpsw** [comment]
Maybe.  Using this sample script is the best starting point https://gopro.github.io/labs/control/dailytl/  It can be modified to support photo timelapse modes, although remember video timelapse is significantly more data efficient.  There is no point to shutting-down between shots if you are externally powering, that just greatly reduces the reliability.  On HERO11 there is a new issue, the UI is interfering by being "helpful", asking at the end of upload if you want clear media (preventing the script from restarting.) So on HERO9/10/Bones this works pretty well, HERO11 needs a fix.

**Reply by serek1107** [reply]
Thanks for reply! I was thinking about using construction timelapse mode with powering off between shots, because I'not yet sure if I will be able to use AC power, so a large battery is more probable. I will also be quite far from gopro so I need an option that consume as little power as it is possible and for example change battery once in a month. I also want to shoot timelapse for 8 hours, upload photos to cloud and then the camera should shut down, for power efficiency. As goes for date efficient I will use 512gb sd card, so I think it won't be a problem to take around 1500 photos a month. 
Do I need to program everything in one QR code, or this setting can be added separately and work together? If it's stupid question sorry for that, I'm completely new in Gopro Labs.

**Reply by dnewman-gpsw** [reply]
Unfortunately it's not likely to be successful.  The wake timer used in Labs is good for about 500 starts before something might go wrong. Adding any cloud connectivity increases the probability of failure.  Best solution is to avoid the camera restarts and avoid cloud connections. Workout how to provide constant power or sufficient batteries such that you can use the camera's default time lapse modes.  Once the camera is going it can run for months. I personally run a video time lapse for 3 months, with AC power.  I've also run a labs style time lapse taking one photo a day for a year, swapping the battery every 45 days.  The idea of running 50 photos a day, over several months is just really stresses reliability.  Labs features are experimental, using a consumer camera not designed for these purposes, extending its capabilities, yes, but with no guarantees.  QR codes can be combined into one or separate actions.

**Reply by serek1107** [reply]
Ok, I'll try to avoid switching on and off, but the connection and auto upload are must have in my situation. Can it be done with photo timelapse mode? If so, can you describe me how to do that? Because in the link from your fist message I see only options of Video Timelapse and Nightlapse video.

**Reply by dnewman-gpsw** [reply]
Labs is for tinkerers to experiment, so do try first. Yes all the required information is on https://gopro.github.io/labs/control/   
mTP is the command that timelapse Photo, that need to be switched in.  Your camera will also need to have a timelapse Photo preset enabled. That is it.
